### DISCUSSION TITLE
No keybinds for show next tab/prev tab available in the new version

### DISCUSSION THREAD

**Discussion by KorigamiK** [zen-browser/desktop#2666] [Ideas]
### Captchas

- [X] I have read the instructions.
- [X] I have searched existing issues and avoided creating duplicates.
- [X] I am not filing an enhancement request.

### What happened?

cannot find these binds and a few others in the new update

![image](https://github.com/user-attachments/assets/525e0dbb-cbd8-4742-8879-0b59e291d892)

### Reproducible?

- [X] I have checked that this issue cannot be reproduced on Mozilla Firefox.

### Version

1.0.1.a.6-1

### What platform are you seeing the problem on?

Linux

**Comment by linear** [comment]
<p><a href="https://linear.app/zen-browser/issue/ZEN-1818/no-keybinds-for-show-next-tabprev-tab-available-in-the-new-version">ZEN-1818 No keybinds for show next tab/prev tab available in the new version</a></p>

**Comment by NocFA** [comment]
Indeed, can replicate.
Tried fresh browser profile & installation.

Previously mentioned in another issue that wasn't directly in reference to this by me & a few others - https://github.com/zen-browser/desktop/issues/1706#issuecomment-2384045695

Hopeful for a fix in good time.

**Comment by saviorand** [comment]
Same here, i think the whole layout in the shortcuts section changed though. Not sure if "navigate back" is the same shortcut

**Comment by NocFA** [comment]
> Same here, i think the whole layout in the shortcuts section changed though. Not sure if "navigate back" is the same shortcut

No, sadly not. 
That instead functions by if you changed tab, back nav would return you to the previous tab you were on. 
Not the same as going forward/backwards relative to the tab you’re on.

**Comment by Microwave-WYB** [comment]
This was the whole reason I switch to zen from firefox, and it's gone?

**Comment by KorigamiK** [comment]
> This was the whole reason I switch to zen from firefox, and it's gone?

Fr regressions are not something we should expect this early in the project

**Comment by omarcresp** [comment]
bumping this, would be great to have this back in zen

**Comment by EuCaue** [comment]
Would love to see this feature make a comeback to Zen! +1

**Comment by ReneEChr** [comment]
+1 getting these shortcuts back, it's a priceless feature

**Comment by ristomatti** [comment]
Alternatively, use e.g. [Keymapper](https://github.com/houmain/keymapper) and fix keybinds of any app.

**Comment by unkindypie** [comment]
+1 on linux

**Comment by GardenerGeralt** [comment]
+1 on linux

**Comment by salamijason** [comment]
Hiya folks. I also agree that this should be a map-able keybind and will be sending this in as a suggestion. In the meantime, I've resorted to using ctrl+pgup/pgdown.

**Comment by KorigamiK** [comment]
@salamijason  please do not close this issue. As previously discussed, this issue is not for a new feature. It is a regression to a previous version (alpha 3) which allowed such a key bind. 

I would love to see this issue in the roadmap again as soon as possible.

**Comment by salamijason** [comment]
Alright, i'll try sending it in. But not sure when the dev will be able to look at it since he's more focused on fixing bugs. A while back, the keyboard shortcuts were completely redone from the ground up and less-used keybinds were left to focus on the more common ones.

**Comment by salamijason** [comment]
Was notified that it's already been added back. Will be released soon.

https://github.com/zen-browser/components/pull/69
https://github.com/zen-browser/l10n-packs/pull/87

**Comment by KorigamiK** [comment]
> Was notified that it's already been added back. Will be released soon.
> 
> 
> 
> https://github.com/zen-browser/components/pull/69
> 
> https://github.com/zen-browser/l10n-packs/pull/87

That hyperlink doesn't seem to work, maybe a typo? can we expect this to be released in a.17 ?

**Reply by RayZ3R0** [reply]
a.18 maybe

**Reply by KorigamiK** [reply]
> a.18 maybe

Aw man need immediately so I can finally move off of a.3 to this

**Reply by salamijason** [reply]
Aw my bad, completely forgot to check if it was visible to non members. But yes, prolly coming in the next release.

Cheff doesn't really do patches or different iterations of versions (e.g. a17-1, a17-2) but don't worry, heard that he may be releasing soon.

**Reply by KorigamiK** [reply]
> Aw my bad, completely forgot to check if it was visible to non members. But yes, prolly coming in the next release.
> 
> Cheff doesn't really do patches or different iterations of versions (e.g. a17-1, a17-2) but don't worry, heard that he may be releasing soon.

Oh looking forward to it! Btw is there a criteria for becoming a member, I'd love to contribute to zen as well?

**Reply by salamijason** [reply]
> is there a criteria for becoming a member

Join the discord and be active there! That's how all of us ended up becoming issue handlers. We all just offered to help with the github issues.

**Reply by KorigamiK** [reply]
@RayZ3R0 @salamijason I tried out a.18-1 and I still don't see these tab actions on the keyboard shortcuts page (which all existed before):

![image](https://github.com/user-attachments/assets/f80d3dca-6d3b-4c76-9566-9c94b36fc358)

**Reply by salamijason** [reply]
Hmm, yeah I checked too and couldn't find them. Really weird. I guess they didn't pull it for this update. Honestly I don't know.

**Reply by KorigamiK** [reply]
> Hmm, yeah I checked too and couldn't find them. Really weird. I guess they didn't pull it for this update. Honestly I don't know.

Yeah, can you tell me who I should tag on the discord to join the zen components development?

**Reply by salamijason** [reply]
No specific person in particular. The devs are active often and you can try mentioning it when they're in the server.

**Comment by fanesz** [comment]
still no update?, it's the most feature that I love from Zen man

**Comment by saviorand** [comment]
@fanesz this works now on the latest version

**Reply by fanesz** [reply]
@saviorand where? my version is up to date: 1.0.1-a.22 (Firefox 132.0.1) (64-bit),
but still can't find any shortcut setting for that

**Reply by saviorand** [reply]
I don't see it in Navigation settings but for me the shortcut is Cmd+option+leftarrow or rightarrow. I remember changing it back when the setting was there in the navigation , so not sure if i hit the weird edge case somehow

**Reply by KorigamiK** [reply]
yea, you can't say it's working, if the shortcuts are not in the settings

**Reply by saviorand** [reply]
@KorigamiK btw did you find someone to tag on Discord to raise this issue up?

**Reply by KorigamiK** [reply]
> @KorigamiK btw did you find someone to tag on Discord to raise this issue up?

I tagged @ maurob. on discord. I suppose he understands the issue but isn't working on it right now

**Reply by vraravam** [reply]
Afaik, these are not included in the above UI since they are hardcoded from FF.
In FF, on macos, I always navigate across the adjacent tabs (cycle through based on tab location/number) using the `Cmd+Opt/Alt+LeftArrow` and the `Cmd+Opt/Alt+RightArrow` key combos. The same works in Zen as well, and I haven't found a way to change them - since they are from the "older" era of horizontal tabs. (In the vertical tabs usage, I would like to swap to using up/down arrow keys in those same combos, but haven't found a way to do that)

**Comment by ahmedYasserM** [comment]
any updates ?

**Reply by KorigamiK** [reply]
For now, a hack to achieve custom shortcuts is using [fx-autoconfig](https://github.com/MrOtherGuy/fx-autoconfig) and creating a user script that kinda works like this;
```js
// ==UserScript==
// @name           Tab Navigation Hotkeys
// @namespace      tab_navigation_hotkeys
// @version        1.0
// @description    Use alt-j and alt-k to navigate tabs
// ==/UserScript==
function key_move_tabs() {
  UC_API.Hotkeys.define({
    modifiers: "alt",
    key: "J",
    id: "key_move_next",
    command: (win) => {
      gBrowser.tabContainer.advanceSelectedTab(1, true);
    },
  }).autoAttach({ suppressOriginalKey: true });

  UC_API.Hotkeys.define({
    modifiers: "alt",
    key: "K",
    id: "key_move_prev",
    command: (win) => {
      gBrowser.tabContainer.advanceSelectedTab(-1, true);
    },
  }).autoAttach({ suppressOriginalKey: true });
}

key_move_tabs();
```

Make sure to clear startup cache for applying the changes

**Comment by Spurkus** [comment]
Really sucks that they haven't included this shortcut back into Zen. This was my favourite shortcut to move across tabs faster. Is there any update on this?

**Reply by Spurkus** [reply]
Found this pull request https://github.com/zen-browser/desktop/pull/3599

**Comment by shivajreddy** [comment]
+1 on linux. Please zen, we need custom keymap settings for next/previous tab

**Comment by furuochen-dev** [comment]
This is NOT fixed.
